src/tokenizer.rs

```diff
@@ -1206,6 +1212,41 @@ impl<'a> Tokenizer<'a> {
         }
     }
 
+    /// Consume the next character, then parse a custom binary operator. The next character should be included in the prefix
+    fn consume_for_binop(
+        &self,
+        chars: &mut State,
+        prefix: &str,
+        default: Token,
+    ) -> Result<Option<Token>, TokenizerError> {
+        chars.next(); // consume the first char
+        self.start_binop(chars, prefix, default)
+    }
+
+    /// parse a custom binary operator
+    fn start_binop(
+        &self,
+        chars: &mut State,
+        prefix: &str,
+        default: Token,
+    ) -> Result<Option<Token>, TokenizerError> {
+        let mut s = prefix.to_string();
+        let mut is_custom_operator = false;
+        while let Some(&ch) = chars.peek() {
+            if !self.dialect.is_custom_operator_part(ch) {
+                break;
+            }
+            s.push(ch);
+            is_custom_operator = true;
+            chars.next();
+        }
+        if is_custom_operator {
+            Ok(Some(Token::CustomBinaryOperator(s)))
+        } else {
+            Ok(Some(default))
+        }
```

Comment:
We could avoid the string allocation for the common case of a non-custom binop, something like:

```suggestion
        let mut custom = None;
        while let Some(&ch) = chars.peek() {
            if !self.dialect.is_custom_operator_part(ch) {
                break;
            }

            custom.get_or_insert_with(|| prefix.to_string()).push(ch);
            chars.next();
        }

        Ok(Some(custom.map(Token::CustomBinaryOperator).unwrap_or(default)))
```

Reply:
Very True ! Let's do it

Reply:
done